third_party/move/tools/move-unit-test/src/test_runner.rs

```diff
@@ -305,7 +295,21 @@ impl SharedTestingConfig {
                 .into(),
         );
         match session.finish_with_extensions() {
-            Ok((cs, extensions)) => (Ok(cs), Ok(extensions), return_result, test_run_info),
+            Ok((cs, mut extensions)) => {
+                match gas_meter.charge_write_set(&cs, &mut extensions)
+                {
+                    Ok(()) => {
+                        // recompute used gas after write_set charging
+                        test_run_info.instructions_executed = Gas::new(self.execution_bound)
+                            .checked_sub(gas_meter.remaining_gas())
+                            .unwrap()
+                            .into();
```

Comment:
This is a bit hacky. I think I understand what you are trying to achieve though.

Instead of letting the unit test framework calculate the amount of gas used or number of instructions executed, we should just let it return the execution result (which should include the custom gas meter) to the client (caller of `run_move_unit_tests_with_gas_meter`) and then they'll be able to render a report in whatever way they like.

This will satisfy all the Aptos use cases and provide more flexibility for people trying to use Move outside of Aptos.